client/src/components/Home.js

```diff
@@ -25,6 +27,32 @@ const Home = () => {
     }
   };
 
+  const handleDeleteRequest = async (id) => {
+    try {
+      const resp = await fetch(`${HOST}/watchlist`, {
+        method: 'DELETE',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ id }),
+      });
+      if (resp.ok) {
+        removeChannel(id);
+        setIsSubscribed(!isSubscribed);
+        console.log('running');
+      }
+    } catch (e) {
+      console.log(e, 'Failed to delete channel');
+    }
+  };
+
+  const removeChannel = (channelId) => {
+    channels = channels.filter((element) => {
```

Comment:
currently this does nothing since react state is immutable. You need to use `setChannels(channels.filter(...))`

Reply:
I believe it should work as intended this time :).

Reply:
💪 . This works but assignment isn't necessary, I made a small clean up commit https://github.com/aziz512/youtube-tracker/pull/47/commits/3501a682e778af480fd89e26cc875777eca969bd .

Reply:
I'm in a rush to push things to finish the powerpoint. Good catch.